File: demo-web-java/src/main/java/com/example/demo/controller/WelcomeController.java
package com.example.demo.controller;

import com.example.demo.security.CustomUserDetails;
import lombok.extern.slf4j.Slf4j;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
@Slf4j
public class WelcomeController {

  @GetMapping("/")
  public String welcome(Model model, @AuthenticationPrincipal CustomUserDetails customUserDetails) {
    String message = "Hello Spring With Java";
    log.debug("customUserDetails:{}" + customUserDetails);
    model.addAttribute("message", message);
    model.addAttribute("currentUserDetail", customUserDetails);
    return "welcome";
  }

}

File: demo-web-scala/src/main/scala/com/example/demo/controller/WelcomeController.scala
package com.example.demo.controller

import com.example.demo.security.CustomUserDetails
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping

@Controller
class WelcomeController {

  val log: Logger = LoggerFactory.getLogger(this.getClass);

  @GetMapping(Array("/"))
  def welcome(model: Model, @AuthenticationPrincipal customUserDetails: CustomUserDetails): String = {
    val message = "Hello Spring With Java"
    log.debug("customUserDetails:{}", customUserDetails)
    model.addAttribute("message", message)
    model.addAttribute("currentUserDetail", customUserDetails)
    "welcome"
  }

}
